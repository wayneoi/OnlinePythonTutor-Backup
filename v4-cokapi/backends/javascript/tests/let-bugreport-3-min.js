// minimized form of let-bugreport-3.js
// weird things starting in step 3:
// - empty-looking undefined variable above 'arg'
// - 'arg (block 1)' is a duplicate of arg, seems like it should be 'start'
// - all sorts of let-defined vars look way off!
const baseRange = function (...arg) {
    let [start, end, step = 1] = arg;
    let index = -1, length = Math.ceil( (end -start) / step );
    console.log('index:', index);
}

baseRange(4,1,-1);
