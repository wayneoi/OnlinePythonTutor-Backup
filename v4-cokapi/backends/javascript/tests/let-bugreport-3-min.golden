{ code: '// minimized form of let-bugreport-3.js\n// weird things starting in step 3:\n// - empty-looking undefined variable above \'arg\'\n// - \'arg (block 1)\' is a duplicate of arg, seems like it should be \'start\'\n// - all sorts of let-defined vars look way off!\nconst baseRange = function (...arg) {\n    let [start, end, step = 1] = arg;\n    let index = -1, length = Math.ceil( (end -start) / step );\n    console.log(\'index:\', index);\n}\n\nbaseRange(4,1,-1);',
  trace: 
   [ { stdout: '',
       func_name: '',
       stack_to_render: [],
       globals: {},
       ordered_globals: [],
       line: 6,
       col: 0,
       event: 'step_line',
       heap: {} },
     { stdout: '',
       func_name: '',
       stack_to_render: [],
       globals: { baseRange: [ 'REF', 1 ] },
       ordered_globals: [ 'baseRange' ],
       line: 12,
       col: 0,
       event: 'step_line',
       heap: 
        { '1': 
           [ 'JS_FUNCTION',
             '',
             'function (...arg) {\n    let [start, end, step = 1] = arg;\n    let index = -1, length = Math.ceil( (end -start) / step );\n    console.log(\'index:\', index);\n}',
             null,
             null ] } },
     { stdout: '',
       func_name: '',
       stack_to_render: 
        [ { func_name: '',
            frame_id: 2,
            is_highlighted: true,
            is_parent: false,
            is_zombie: false,
            parent_frame_id_list: [],
            unique_hash: '_f2',
            ordered_varnames: 
             [ 'arg',
               'start (block 1)',
               'end (block 1)',
               'step (block 1)',
               'index (block 1)',
               'length (block 1)' ],
            encoded_locals: 
             { arg: [ 'REF', 2 ],
               'start (block 1)': [ 'JS_SPECIAL_VAL', 'undefined' ],
               'end (block 1)': [ 'JS_SPECIAL_VAL', 'undefined' ],
               'step (block 1)': [ 'JS_SPECIAL_VAL', 'undefined' ],
               'index (block 1)': [ 'JS_SPECIAL_VAL', 'undefined' ],
               'length (block 1)': [ 'JS_SPECIAL_VAL', 'undefined' ] } } ],
       globals: { baseRange: [ 'REF', 1 ] },
       ordered_globals: [ 'baseRange' ],
       line: 7,
       col: 4,
       event: 'call',
       heap: 
        { '1': 
           [ 'JS_FUNCTION',
             '',
             'function (...arg) {\n    let [start, end, step = 1] = arg;\n    let index = -1, length = Math.ceil( (end -start) / step );\n    console.log(\'index:\', index);\n}',
             null,
             null ],
          '2': [ 'LIST', 4, 1, -1 ] } },
     { stdout: '',
       func_name: '',
       stack_to_render: 
        [ { func_name: '',
            frame_id: 2,
            is_highlighted: true,
            is_parent: false,
            is_zombie: false,
            parent_frame_id_list: [],
            unique_hash: '_f2',
            ordered_varnames: 
             [ 'arg',
               'start (block 1)',
               'end (block 1)',
               'step (block 1)',
               'index (block 1)',
               'length (block 1)' ],
            encoded_locals: 
             { arg: [ 'REF', 2 ],
               'start (block 1)': [ 'JS_SPECIAL_VAL', 'undefined' ],
               'end (block 1)': [ 'JS_SPECIAL_VAL', 'undefined' ],
               'step (block 1)': [ 'JS_SPECIAL_VAL', 'undefined' ],
               'index (block 1)': [ 'JS_SPECIAL_VAL', 'undefined' ],
               'length (block 1)': [ 'JS_SPECIAL_VAL', 'undefined' ] } } ],
       globals: { baseRange: [ 'REF', 1 ] },
       ordered_globals: [ 'baseRange' ],
       line: 7,
       col: 9,
       event: 'step_line',
       heap: 
        { '1': 
           [ 'JS_FUNCTION',
             '',
             'function (...arg) {\n    let [start, end, step = 1] = arg;\n    let index = -1, length = Math.ceil( (end -start) / step );\n    console.log(\'index:\', index);\n}',
             null,
             null ],
          '2': [ 'LIST', 4, 1, -1 ] } },
     { stdout: '',
       func_name: '',
       stack_to_render: 
        [ { func_name: '',
            frame_id: 2,
            is_highlighted: true,
            is_parent: false,
            is_zombie: false,
            parent_frame_id_list: [],
            unique_hash: '_f2',
            ordered_varnames: 
             [ 'arg',
               'start (block 1)',
               'end (block 1)',
               'step (block 1)',
               'index (block 1)',
               'length (block 1)' ],
            encoded_locals: 
             { arg: [ 'REF', 2 ],
               'start (block 1)': 4,
               'end (block 1)': [ 'JS_SPECIAL_VAL', 'undefined' ],
               'step (block 1)': [ 'JS_SPECIAL_VAL', 'undefined' ],
               'index (block 1)': [ 'JS_SPECIAL_VAL', 'undefined' ],
               'length (block 1)': [ 'JS_SPECIAL_VAL', 'undefined' ] } } ],
       globals: { baseRange: [ 'REF', 1 ] },
       ordered_globals: [ 'baseRange' ],
       line: 7,
       col: 16,
       event: 'step_line',
       heap: 
        { '1': 
           [ 'JS_FUNCTION',
             '',
             'function (...arg) {\n    let [start, end, step = 1] = arg;\n    let index = -1, length = Math.ceil( (end -start) / step );\n    console.log(\'index:\', index);\n}',
             null,
             null ],
          '2': [ 'LIST', 4, 1, -1 ] } },
     { stdout: '',
       func_name: '',
       stack_to_render: 
        [ { func_name: '',
            frame_id: 2,
            is_highlighted: true,
            is_parent: false,
            is_zombie: false,
            parent_frame_id_list: [],
            unique_hash: '_f2',
            ordered_varnames: 
             [ 'arg',
               'start (block 1)',
               'end (block 1)',
               'step (block 1)',
               'index (block 1)',
               'length (block 1)' ],
            encoded_locals: 
             { arg: [ 'REF', 2 ],
               'start (block 1)': 4,
               'end (block 1)': 1,
               'step (block 1)': [ 'JS_SPECIAL_VAL', 'undefined' ],
               'index (block 1)': [ 'JS_SPECIAL_VAL', 'undefined' ],
               'length (block 1)': [ 'JS_SPECIAL_VAL', 'undefined' ] } } ],
       globals: { baseRange: [ 'REF', 1 ] },
       ordered_globals: [ 'baseRange' ],
       line: 7,
       col: 21,
       event: 'step_line',
       heap: 
        { '1': 
           [ 'JS_FUNCTION',
             '',
             'function (...arg) {\n    let [start, end, step = 1] = arg;\n    let index = -1, length = Math.ceil( (end -start) / step );\n    console.log(\'index:\', index);\n}',
             null,
             null ],
          '2': [ 'LIST', 4, 1, -1 ] } },
     { stdout: '',
       func_name: '',
       stack_to_render: 
        [ { func_name: '',
            frame_id: 2,
            is_highlighted: true,
            is_parent: false,
            is_zombie: false,
            parent_frame_id_list: [],
            unique_hash: '_f2',
            ordered_varnames: 
             [ 'arg',
               'start (block 1)',
               'end (block 1)',
               'step (block 1)',
               'index (block 1)',
               'length (block 1)' ],
            encoded_locals: 
             { arg: [ 'REF', 2 ],
               'start (block 1)': 4,
               'end (block 1)': 1,
               'step (block 1)': -1,
               'index (block 1)': [ 'JS_SPECIAL_VAL', 'undefined' ],
               'length (block 1)': [ 'JS_SPECIAL_VAL', 'undefined' ] } } ],
       globals: { baseRange: [ 'REF', 1 ] },
       ordered_globals: [ 'baseRange' ],
       line: 8,
       col: 4,
       event: 'step_line',
       heap: 
        { '1': 
           [ 'JS_FUNCTION',
             '',
             'function (...arg) {\n    let [start, end, step = 1] = arg;\n    let index = -1, length = Math.ceil( (end -start) / step );\n    console.log(\'index:\', index);\n}',
             null,
             null ],
          '2': [ 'LIST', 4, 1, -1 ] } },
     { stdout: '',
       func_name: '',
       stack_to_render: 
        [ { func_name: '',
            frame_id: 2,
            is_highlighted: true,
            is_parent: false,
            is_zombie: false,
            parent_frame_id_list: [],
            unique_hash: '_f2',
            ordered_varnames: 
             [ 'arg',
               'start (block 1)',
               'end (block 1)',
               'step (block 1)',
               'index (block 1)',
               'length (block 1)' ],
            encoded_locals: 
             { arg: [ 'REF', 2 ],
               'start (block 1)': 4,
               'end (block 1)': 1,
               'step (block 1)': -1,
               'index (block 1)': -1,
               'length (block 1)': 3 } } ],
       globals: { baseRange: [ 'REF', 1 ] },
       ordered_globals: [ 'baseRange' ],
       line: 9,
       col: 4,
       event: 'step_line',
       heap: 
        { '1': 
           [ 'JS_FUNCTION',
             '',
             'function (...arg) {\n    let [start, end, step = 1] = arg;\n    let index = -1, length = Math.ceil( (end -start) / step );\n    console.log(\'index:\', index);\n}',
             null,
             null ],
          '2': [ 'LIST', 4, 1, -1 ] } },
     { stdout: 'index: -1\n',
       func_name: '',
       stack_to_render: 
        [ { func_name: '',
            frame_id: 2,
            is_highlighted: true,
            is_parent: false,
            is_zombie: false,
            parent_frame_id_list: [],
            unique_hash: '_f2',
            ordered_varnames: [ 'arg', '__return__' ],
            encoded_locals: 
             { arg: [ 'REF', 2 ],
               __return__: [ 'JS_SPECIAL_VAL', 'undefined' ] } } ],
       globals: { baseRange: [ 'REF', 1 ] },
       ordered_globals: [ 'baseRange' ],
       line: 9,
       col: 0,
       event: 'return',
       heap: 
        { '1': 
           [ 'JS_FUNCTION',
             '',
             'function (...arg) {\n    let [start, end, step = 1] = arg;\n    let index = -1, length = Math.ceil( (end -start) / step );\n    console.log(\'index:\', index);\n}',
             null,
             null ],
          '2': [ 'LIST', 4, 1, -1 ] } },
     { stdout: 'index: -1\n',
       func_name: '',
       stack_to_render: [],
       globals: { baseRange: [ 'REF', 1 ] },
       ordered_globals: [ 'baseRange' ],
       line: 12,
       col: 18,
       event: 'step_line',
       heap: 
        { '1': 
           [ 'JS_FUNCTION',
             '',
             'function (...arg) {\n    let [start, end, step = 1] = arg;\n    let index = -1, length = Math.ceil( (end -start) / step );\n    console.log(\'index:\', index);\n}',
             null,
             null ] } } ] }
